pkgs/development/python-modules/deepwave/default.nix

```diff
@@ -0,0 +1,52 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pytorch
+, ninja
+, scipy
+, which
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "deepwave";
+  version = "0.0.11";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "ar4";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-d4EahmzHACHaeKoNZy63OKwWZdlHbUydrbr4fD43X8s=";
+  };
+
+  # unable to find ninja although it is available
+  postPatch = ''
+    substituteInPlace setup.cfg --replace "ninja" ""
+  '';
+
+  # The source files are compiled at runtime and cached at the
+  # $HOME/.cache folder, so for the check phase it is needed to
+  # have a temporary home
+  preBuild = ''
+    export HOME=$(mktemp -d)
+  '';
+
+  propagatedBuildInputs = [ pytorch ];
+
+  checkInputs = [
+    ninja
```

Comment:
Shouldn't this be in nativeBuildInputs? Then maybe the postPatch can be changed too.

Reply:
it only uses the ninja at runtime, so it will not be used in the buildPhase, hence no need to put it in the nativeBuildInputs. The reason I did the postPatch thing is because the setup.cfg most likely asks for the pip version of ninja, which we don't have. In the first time you import the library it compiles everything with ninja and caches the binaries.